xen/drivers/passthrough/iommu.c

```diff
@@ -249,22 +249,39 @@ void iommu_domain_destroy(struct domain *d)
     arch_iommu_domain_destroy(d);
 }
 
-int iommu_map_page(struct domain *d, unsigned long gfn, unsigned long mfn,
-                   unsigned int flags)
+int iommu_map_pages(struct domain *d, unsigned long gfn, unsigned long mfn,
+                    unsigned long page_count, unsigned int flags)
 {
     const struct domain_iommu *hd = dom_iommu(d);
-    int rc;
+    int rc = 0;
```

Comment:
The proper solution has been found. I won't check for hd->platform_ops->map_page/hd->platform_ops->unmap_page to be present as it was done before my changes. So, -ENOSYS is not needed here.